storage/fanout.go

```diff
@@ -210,7 +233,9 @@ func NewMergeQuerier(queriers []Querier) Querier {
 		return filtered[0]
```

Comment:
This isn't including the primaryQuerier

Reply:
The input `queriers` still includes the primary querier, and subsequently so does filtered. primaryQuerier is just passed in addition to be able to identify the primaryQuerier